test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/EdmAnnotationsTargetTests.cs

```diff
@@ -0,0 +1,862 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="EdmAnnotationsTargetTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+#if NETCOREAPP3_1
+using System.Text.Encodings.Web;
+using System.Text.Json;
+#endif
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.OData.Edm.Csdl;
+using Microsoft.OData.Edm.Validation;
+using Microsoft.OData.Edm.Vocabularies;
+using Microsoft.OData.Edm.Vocabularies.V1;
+using Xunit;
+
+namespace Microsoft.OData.Edm.Tests.Csdl
+{
+    public class EdmAnnotationsTargetTests
+    {
+        #region Validations
+
+        [Fact]
+        public void InitializeEdmAnnotationsTargetWithNull()
+        {
+            Action parseAction = () => new EdmAnnotationsTarget(null);
+
+            Assert.Throws<ArgumentNullException>(parseAction);
+        }
+
+        [Fact]
+        public void InitializeEdmAnnotationsTarget_FirstSegmentIsNotContainer()
+        {
+            // Arrange
+            EdmModel model = new EdmModel();
+            EdmEntityType customer = new EdmEntityType("NS", "Customer");
+            customer.AddKeys(customer.AddStructuralProperty("Id", EdmCoreModel.Instance.GetInt32(false)));
+            customer.AddStructuralProperty("Name", EdmPrimitiveTypeKind.String, isNullable: false);
+            model.AddElement(customer);
+            IEdmProperty nameProperty = customer.DeclaredProperties.Where(x => x.Name == "Name").FirstOrDefault();

```

Comment:
try to use Mock.

Mock<IEdmEntityType> customer = new Mock<IEdmEntityType>();
Mock<IEdmProperty> nameProperty = new Mock<IEdmProperty>();

Action test = () => new EdmAnntationsTarget(customer.Object, namePropert.Object);

...

Reply:
I don't see alot of value installing the `Moq` package for these tests